### PR TITLE
Insert 

### PR DESCRIPTION
**Pull request by naomijub**
Todos:
- [x] Simples `Insert` 
- [x] `Insert`  with uuid
- [x] Char
- [x] Integer - WIP solution
- [x] String
- [x] Uuid
- [x] Float
- [x] Boolean
- [x] Vector
- [x] Map
- [x] Precise
- [x] DateTime
- [x] Nil

### CODE REVIEW

File: src/model/types.rs [hunk 1]
```diff
@@ -197,3 +217,87 @@ impl From<HashMap<String, Types>> for Types {
...
+
+    use super::*;
+
+    #[test]
+    fn parse_map_with_uuid() {
+        assert_eq!(
+            Ok((
+                "",
+                Types::Map(
+                    vec![(
```
Comment (otaviopace) [inline src/model/types.rs:233]: Idk if it's worth bringing this crate to the table, but it may help making it simpler that you're creating a `HashMap` without having to use a `Vec`: https://github.com/GrayJack/sugars

File: src/parser/types.rs [hunk 2]
```diff
@@ -38,15 +42,65 @@ pub fn set(input: &str) -> IResult<&str, Vec<String>, VerboseError<&str>> {
...
+}
+
+pub fn vector(input: &str) -> IResult<&str, Vec<Types>, VerboseError<&str>> {
+    context(
+        "vector",
+        preceded(
+            tag("["),
+            cut(terminated(
+                separated_list0(preceded(char(','), sp), wql_value),
+                alt((char(']'), preceded(sp, char(']')))),
```
Comment (otaviopace) [inline src/parser/types.rs:81]: Why is this needed (two `]`)?

**Review by otaviopace — APPROVED** [review]
LGTM

I mean, I approved before it's done 😄

File: src/parser/types.rs (hunk 2 above)
Comment (naomijub) [reply src/parser/types.rs:81]: `alt` gives an alternative parser choice, `char(])` looks only for a terminating element of char `]`, while `preceded` with `sp`(spaces) looks for a any terminating element consisting of whitespaces and a char `]`.

Ex:
- only `char(']')` would parse this `[1, 2, 3, 4]` but not `[1, 2, 3, 4 ]` or `[1, 2, 3, 4.         ]`
- only `preceded(sp, char(']'))` would parse `[1, 2, 3, 4 ]` or `[1, 2, 3, 4.         ]` but sometimes it bugged with `[1, 2, 3, 4]`

**Comment by naomijub** [comment]
It is faaaaar from done
